src/App.jsx

```diff
@@ -1,17 +1,40 @@
 import './App.css';
+import messages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
+
+function App() {
+  const [chatData, setChatData] = useState(messages);
+  const [likesCount, setLikesCount] = useState(0);
+
+  const toggleLiked = (chatEntryId) => {
+    setChatData(chats => {
+      return chats.map(chat => {
+        if (chat.id === chatEntryId) {
+          return { ...chat, liked: !chat.liked };
+        } else {
+          return chat;
+        }
+      });
+    });
+  };
+
+  const calculateLikes = (isLiked) => {
+    setLikesCount((likesCount) => isLiked ? likesCount - 1 : likesCount + 1);
+  };
```

Comment:
With the change listed above we don't need this and we won't have worry about passing it down as a prop to handle another piece of state.